Realm/RLMResults.h

```diff
@@ -408,6 +427,173 @@ __attribute__((warn_unused_result));
+
+ This method opens a write transaction that creates or updates a subscription.
+ It's advised to *not* loop over this method in order to create multiple subscriptions at once.
+ This could create a performance bottleneck by opening multiple unnecessary write transactions.
+ @see: `[RLMSyncSubscription update:queue:onComplete:]` in order to create multiple subscriptions.
+
+ @param name The name used  to identify the subscription.
+ @param waitForSyncMode Dictates when the completion handler is called
+ @param queue The queue where the completion dispatches.
+ @param completion The block called after the subscription download is
+ completed. Return behavior is dictated by the RLMWaitForSyncMode.
+ @see ``RLMWaitForSyncMode``
+ */
+- (void)subscribeWithName:(NSString *_Nullable)name
+              waitForSync:(RLMWaitForSyncMode)waitForSyncMode
+                  onQueue:(dispatch_queue_t _Nullable)queue
+               completion:(RLMResultsCompletionBlock)completion
+__attribute__((swift_attr("@_unsafeInheritExecutor")));
+
+/**
+ Creates an RLMSyncSubscription matching the RLMResults' local filter.
+
+ After committing the subscription to the realm's local subscription set, the method
+ will wait for downloads according to the `RLMWaitForSyncMode`.
+ @see ``RLMWaitForSyncMode``
+
+ If `subscribeWithName` is called without a name on a query
+ that's already subscribed to without a name, another subscription is not created.
+ If `subscribeWithName` is called without a name on a query
+ that's already subscribed to with a name, an additional subscription is created without a name.
+ If `subscribeWithName` is called with a name on a query that's
+ already subscribed to without a name, an additional subscription is created
+ with the provided name.
+ If `subscribeWithName` is called with a name that's in use on
+ a different query, the old subscription is updated with the new query.
+ If `subscribeWithName` is called with the same name and
+ same query of an existing subscription, no new subscription is created
+
+ This method opens a write transaction that creates or updates a subscription.
+ It's advised to *not* loop over this method in order to create multiple subscriptions at once.
+ This could create a performance bottleneck by opening multiple unnecessary write transactions.
+ @see: `[RLMSyncSubscription update:queue:onComplete:]` in order to create multiple subscriptions.
+
+ @param name The name used  to identify the subscription.
+ @param waitForSyncMode Dictates when the completion handler is called
+ @param queue The queue where the completion dispatches.
+ @param timeout A timeout which ends waiting for downloads
+ via the completion handler. If the timeout is exceeded the completion
+ handler returns an `RLMErrorClientTimeout`.
+ @see ``RLMErrorClientTimeout``
+ @param completion The block called after the subscription download is
+ completed. Return behavior is dictated by the RLMWaitForSyncMode.
```

Comment:
I'm not sure what "Return behavior is dictated by the RLMWaitForSyncMode" means. I think it's trying to say that when the callback is invoked depends on the mode?